src/frontend/fields/pill/pill-field.vue

```diff
@@ -0,0 +1,198 @@
+                height="8"
+                viewBox="0 0 8 8"
+                fill="none"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path
+                  d="M6.64645 0.646447C6.84171 0.451184 7.15822 0.451185 7.35348 0.646447C7.54874 0.841709 7.54874 1.15822 7.35348 1.35348L4.70699 3.99996L7.35348 6.64645L7.38766 6.68453C7.54783 6.88092 7.53654 7.17042 7.35348 7.35348C7.17042 7.53654 6.88092 7.54783 6.68453 7.38766L6.64645 7.35348L3.99996 4.70699L1.35348 7.35348C1.15822 7.54874 0.841709 7.54874 0.646447 7.35348C0.451185 7.15822 0.451184 6.84171 0.646447 6.64645L3.29293 3.99996L0.646447 1.35348L0.612267 1.31539C0.452092 1.119 0.463389 0.829504 0.646447 0.646447C0.829504 0.463389 1.119 0.452092 1.31539 0.612267L1.35348 0.646447L3.99996 3.29293L6.64645 0.646447Z"
+                  fill="currentColor"
+                />
+              </svg>
+            </button>
+          </span>
+        </div>
+        <slot name="input">
+          <input
+            v-if="!isReadOnly && showInput"
+            ref="inputRef"
+            :value="inputValue"
+            type="text"
+            :name="field.label"
+            class="block py-1 pl-0 text-sm font-normal leading-5 text-gray-900 bg-white rounded-r-md border-0 focus:border-transparent focus:outline-none focus:ring-2 focus:ring-white"
+            :class="
+              pills.length > 0
+                ? 'ltr:ml-1 ltr:pr-0'
+                : 'ltr:ml-1 ltr:pr-0 rtl:mr-1 rtl:pl-0'
+            "
+            @input="handleInputEvent"
+            @keydown.enter.stop="handleEnterKey"
+            @keydown.arrow-down.prevent="handleArrowDown"
+            @keydown.arrow-up.prevent="handleArrowUp"
+            @keydown.escape="handleEscape"
+            @focus="handleFocus"
+            @blur="handleBlur"
+          />
+        </slot>
+      </div>
+      <slot name="dropdown"></slot>
+    </div>
+    <p v-if="hasError" class="text-sm text-error">{{ errors[0] }}</p>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue';
+import type { PropType } from 'vue';
+import type { FieldSpec } from '../../../types';
+import { commonProps } from '../../shared/helpers';
+
+const props = defineProps({
+  ...commonProps,
+
+  pills: {
+    type: Array as PropType<string[]>,
+    required: true,
+  },
+
+  pillBgColor: {
+    type: String,
+    default: 'bg-[#E0F2FE]',
+  },
+
+  pillTextColor: {
+    type: String,
+    default: 'text-blue-800',
+  },
+
+  displayText: {
+    type: Function as PropType<(pill: string) => string>,
+    default: (pill: string) => pill,
+  },
+
+  errors: {
+    type: Array as PropType<string[]>,
+    default: () => [],
+  },
+
+  inputValue: {
+    type: String,
+    default: '',
+  },
+
+  showInput: {
+    type: Boolean,
+    default: true,
+  },
+});
+
+const emit = defineEmits<{
+  add: [value: string];
+  remove: [value: string];
+  'update:inputValue': [value: string];
```

Comment:
Vue events are case sensitive and there is a mismatch here. How about simply:

```suggestion
  'update': [value: string];
```